include/zephyr/net/net_pkt.h

```diff
@@ -836,6 +836,20 @@ static inline void net_pkt_set_ipv6_fragment_id(struct net_pkt *pkt,
 }
 #endif /* CONFIG_NET_IPV6_FRAGMENT */
 
+static inline bool net_pkt_is_ip_reassembled(struct net_pkt *pkt)
+{
+	if ((IS_ENABLED(CONFIG_NET_IPV4_FRAGMENT) &&
+	     net_pkt_family(pkt) == AF_INET &&
+	     net_pkt_ipv4_fragment_more(pkt)) ||
```

Comment:
If its re-assembled won't the more_frags be 0?

Reply:
I was puzzled about this part of code (note, that I've just moved it from other place) as well, but it turned out, that the fragmentation code re-uses the first fragment packet as a base for the reassembled one, so the more flag (for Ipv6) or frag offset (for IPv6) would be only on those reassembled packets (and only on them).

Reply:
Ah, I thought those flags would be reset as once re-assembled they are of no use, but https://datatracker.ietf.org/doc/html/rfc791 doesn't say so, so, I guess it should work.

But this also means if someone uses that helper on each-fragment then it would falsely returns true, encording `fragment_offset == 0 && more_frags == 1` might mitigate this, then its a question of first fragment or reassembled packet.

Reply:
I can add an explicit flag to `net_pkt` structure to indicate that packet is IP reassembled. There should be no ambiguity then.

Reply:
Ah, so, the L2 driver should set this flag? If yes, that should work, as the stack has no way of knowing its 1st fragment or re-assembled frame.

Reply:
No, L2 has nothing to do with fragmentation. The flag would be off by default, and only set if the IP layer reassembled the packet.

Reply:
Ok, I guess we needn't worry about frag-reassembly offload.